envoy/network/dns_resolver.h

```diff
@@ -26,6 +26,20 @@ class DnsResolverFactory : public Config::TypedFactory {
       const envoy::config::core::v3::TypedExtensionConfig& typed_dns_resolver_config) const PURE;
 
   std::string category() const override { return std::string(DnsResolverCategory); }
+
+  /*
+   * Initialize the related data for this type of DNS resolver.
+   * For some DNS resolvers, like c-ares, there are some specific data structure
+   * needs to be initialized before using it to resolve target.
+   */
+  virtual void init() {}
+
+  /*
```

Comment:
Use `/**` per Doxygen style, here and above.

Reply:
nit: can you rename them initialize() and terminate() as we tend to avoid abbreviations like `init` and `terminate` or `shutdown` is more of the dual of `initialize` then `cleanup`.

Reply:
typically you respond "ACK" when the comment was not actionable. If you are actually doing what is asked you'd respond "Done" or just resolve the comment.

It looks like you actually did this.